pkg/configauditreport/controller/controller.go

```diff
@@ -105,6 +101,29 @@ func (r *ResourceController) SetupWithManager(mgr ctrl.Manager) error {
 		{kind: kube.KindLimitRange, forObject: &corev1.LimitRange{}, ownsObject: &v1alpha1.ConfigAuditReport{}},
 	}
 
+	// Determine which Kubernetes workloads the controller will reconcile and add them to resources
+	targetWorkloads := r.Config.GetTargetWorkloads()
+	for _, tw := range targetWorkloads {
+		switch tw {
+		case "pod":
+			resources = append(resources, resource{kind: kube.KindPod, forObject: &corev1.Pod{}, ownsObject: &v1alpha1.ConfigAuditReport{}})
+		case "replicaset":
+			resources = append(resources, resource{kind: kube.KindReplicaSet, forObject: &appsv1.ReplicaSet{}, ownsObject: &v1alpha1.ConfigAuditReport{}})
+		case "replicationcontroller":
+			resources = append(resources, resource{kind: kube.KindReplicationController, forObject: &corev1.ReplicationController{}, ownsObject: &v1alpha1.ConfigAuditReport{}})
+		case "statefulset":
+			resources = append(resources, resource{kind: kube.KindStatefulSet, forObject: &appsv1.StatefulSet{}, ownsObject: &v1alpha1.ConfigAuditReport{}})
+		case "daemonset":
+			resources = append(resources, resource{kind: kube.KindDaemonSet, forObject: &appsv1.DaemonSet{}, ownsObject: &v1alpha1.ConfigAuditReport{}})
+		case "cronjob":
+			resources = append(resources, resource{kind: kube.KindCronJob, forObject: r.ObjectResolver.GetSupportedObjectByKind(kube.KindCronJob), ownsObject: &v1alpha1.ConfigAuditReport{}})
+		case "job":
+			resources = append(resources, resource{kind: kube.KindJob, forObject: &batchv1.Job{}, ownsObject: &v1alpha1.ConfigAuditReport{}})
+		default:
+			return fmt.Errorf("workload %s is not supported", tw)
+		}
+	}
+
```

Comment:
could we extract it to a method(pkg-->kube-->object) so we could reuse it for all controllers, wdyt ?

Reply:
Yes, good idea. I have abstracted that into its own [method](https://github.com/aquasecurity/trivy-operator/pull/523/files#diff-7820ee8c3e9b7e90a347b7fe2e1f81faa3933252bb22d1a272e890490f215072). Let me know if that looks good.